src/Operation/FindOneAndUpdate.php

```diff
@@ -59,9 +59,9 @@ public function __construct($databaseName, $collectionName, $filter, $update, ar
             throw new InvalidArgumentTypeException('$update', $update, 'array or object');
         }
 
-        if ( ! \MongoDB\is_first_key_operator($update)) {
-            throw new InvalidArgumentException('First key in $update argument is not an update operator');
-        }
+      //  if ( ! \MongoDB\is_first_key_operator($update)) {
+       //     throw new InvalidArgumentException('First key in $update argument is not an update operator');
```

Comment:
I asked @billybanfield to push his wip branch early just so we have a PR available to look at. This is not yet worth the time to review.